packages/components/src/draggable/index.native.js

```diff
@@ -77,36 +78,59 @@ const Draggable = ( { children, onDragEnd, onDragOver, onDragStart } ) => {
 		}
 	);
 
+	function getFirstTouchEvent( event ) {
+		'worklet';
+
+		return event.allTouches.find(
+			( touch ) => touch.id === currentFirstTouchId.value
+		);
+	}
+
 	const panGesture = Gesture.Pan()
 		.manualActivation( true )
 		.onTouchesDown( ( event ) => {
-			const { x = 0, y = 0 } = event.allTouches[ 0 ];
-			initialPosition.x.value = x;
-			initialPosition.y.value = y;
+			if ( ! currentFirstTouchId.value ) {
+				const firstEvent = event.allTouches[ 0 ];
+				const { x = 0, y = 0 } = firstEvent;
+
+				currentFirstTouchId.value = firstEvent.id;
+
+				initialPosition.x.value = x;
+				initialPosition.y.value = y;
+			}
 		} )
-		.onTouchesMove( ( _, state ) => {
+		.onTouchesMove( ( event, state ) => {
 			if ( ! isPanActive.value && isDragging.value ) {
 				isPanActive.value = true;
 				state.activate();
 			}
-		} )
-		.onUpdate( ( event ) => {
-			lastPosition.x.value = event.x;
-			lastPosition.y.value = event.y;
 
-			if ( onDragOver ) {
-				onDragOver( event );
+			if ( isPanActive.value && isDragging.value ) {
+				const firstEvent = getFirstTouchEvent( event );
+
+				if ( ! firstEvent ) {
+					state.end();
+					return;
```

Comment:
If other touches happen and the first touch is no longer present, we cancel the dragging gesture.